### PR TITLE
add firebase publisher to TFX pipeline

### PR DESCRIPTION
**Pull request by deep-diver**
This Pull Request contains the following changes & prerequisites
- FIREBASE_CREDENTIAL_PATH, FIREBASE_GCS_BUCKET has to be set, and FIREBASE_CREDENTIAL_PATH is a GCS path
- code cell to generate firebase_publisher.py is included
- at the end of training for mobilenet, convert the model to TFLite format and save it to the designated GCS path instead of just `model.save()`
- replaced the final pipeline screenshot

### CODE REVIEW

File: notebooks/Custom_Model_TFX.ipynb [hunk 1]
```diff
@@ -1,1339 +1,1839 @@
 {
```
Comment (sayakpaul) [inline notebooks/Custom_Model_TFX.ipynb:1]: <p>It'd a be good idea to write a comment to guide the users about how to obtain this <code>.json</code>.  I suppose <code>FIREBASE_GCS_BUKCET</code>  needs to be appended <code>appspot.com</code>?  If it's the case then make a note about that too.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/sayakpaul/Dual-Deployments-on-Vertex-AI/pull/4/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='13'/>

Comment (sayakpaul) [inline notebooks/Custom_Model_TFX.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #99.        tflite_model = converter.convert()
</pre><p> </p><p>With this conversion, the model size won't decrease as at all i.e. there won't be any quantization (<a href="https://www.tensorflow.org/lite/performance/post_training_quant" target="_blank">source</a>). I would suggest adding one more line:</p><p> </p><pre class="ql-syntax" spellcheck="false">converter = tf.lite.TFLiteConverter.from_keras_model(model)
converter.optimizations = [tf.lite.Optimize.DEFAULT]
tflite_model =&nbsp;converter.convert()
</pre>
<br/>

 _Reply via <a href='https://app.reviewnb.com/sayakpaul/Dual-Deployments-on-Vertex-AI/pull/4/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='17'/>

Comment (sayakpaul) [inline notebooks/Custom_Model_TFX.ipynb:1]: <p>If some imports are unused then discard them. Also format the component modules with <code>black</code>  which is what I also did. This way the linings are beautifully formatted. I'd also log the TFLite model URI with <code>logging.info()</code>.  There's a small typo in: <code>logging.info(f"(MY)doanload model.tflite is completed")</code>.</p><p> <span class="ql-cursor">﻿</span></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/sayakpaul/Dual-Deployments-on-Vertex-AI/pull/4/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='23'/>

Comment (sayakpaul) [inline notebooks/Custom_Model_TFX.ipynb:1]: <p>Clear this particular output.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/sayakpaul/Dual-Deployments-on-Vertex-AI/pull/4/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='30'/>

**Comment by deep-diver** [comment]
@sayakpaul 

please review the modified notebook once more

**Review by sayakpaul — APPROVED** [review]

**Pull request merged**